lib/permissions.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const ADMIN = 'ADMIN';
+const REVIVE_FEATURE = 'REVIVE_FEATURE';
+const CREATE_FEATURE = 'CREATE_FEATURE';
+const UPDATE_FEATURE = 'UPDATE_FEATURE';
+const DELETE_FEATURE = 'DELETE_FEATURE';
+const CREATE_STRATEGY = 'CREATE_STRATEGY';
+const UPDATE_STRATEGY = 'UPDATE_STRATEGY';
+const DELETE_STRATEGY = 'DELETE_STRATEGY';
+const UPDATE_APPLICATION = 'UPDATE_APPLICATION';
+
+function requirePerms(prms) {
+    return (req, res, next) => {
+        for (const permission of prms) {
+            if (
+                req.user &&
+                req.user.permissions &&
+                (req.user.permissions.indexOf(ADMIN) !== -1 ||
+                    req.user.permissions.indexOf(permission) !== -1)
+            ) {
+                return next();
+            }
+        }
+        return res
+            .status(403)
```

Comment:
TODO: frontend needs to handle 403 errors so that the user is formally informed about the missing permission. It would also be nice if the error message also states which permission it expects.